### ISSUE TITLE
I had to force set hostName to get httpd to work

### ISSUE THREAD

**Issue by jappeace** [hercules-ci/arion#198]
```
  config.services.webserver.nixos.configuration.networking.hostName = lib.mkForce "treesweb";
```

If I didn't do this httpd would crash because no [servername](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/web-servers/apache-httpd/default.nix#L289) was set.

**Comment by roberth** [comment]
> `config.services.webserver.service.hostname`

This only has a `null` default so can be set.

> `onfig.services.webserver.nixos.configuration.networking.hostName`

This seems to be the troublemaker. It is set to `""` unless you force it. I suspect that this odd empty value has to do with adopting the hostname provided by the container runtime. NixOS support and container hostname support were merged 2 days apart but were the result of different authors, so I think it just wasn't thought through at the time. So right now there's no real connection between the two options. We could make the one set the other. What do you think?

**Comment by jappeace** [comment]
yes, I'd expect `config.services.webserver.service.hostname` to be set to `webserver` since that's the name of the `<name>`, and I also expect `config.services.webserver.nixos.configuration.networking.hostName` to mirror that value.

**Comment by jappeace** [comment]
I don't think this is a high priority issue however, the other [one](https://github.com/hercules-ci/arion/issues/197) was annoying, this issue I can live with :)
